### PR TITLE
Change tx_seq -> tx_digest of EventID

### PR DESCRIPTION
**Pull request by xydas97**

### CODE REVIEW

File: crates/sui-indexer/src/models/events.rs [hunk 1]
```diff
@@ -28,7 +28,7 @@ pub struct Event {
 #[diesel(table_name = events)]
 pub struct NewEvent {
     pub transaction_digest: Option<String>,
-    pub transaction_sequence: i64,
+    //pub transaction_sequence: i64,
```
Comment (patrickkuo) [inline crates/sui-indexer/src/models/events.rs:31]: we should remove this line

File: crates/sui-indexer/src/models/events.rs [hunk 2]
```diff
@@ -73,9 +73,10 @@ pub fn event_to_new_event(e: SuiEventEnvelope) -> Result<NewEvent, IndexerError>
             e.timestamp
         ))
     })?;
+
     Ok(NewEvent {
         transaction_digest: e.tx_digest.map(|digest| format!("{:?}", digest)),
-        transaction_sequence: e.id.tx_seq,
+        //transaction_sequence: e.id.tx_seq,
```
Comment (patrickkuo) [inline crates/sui-indexer/src/models/events.rs:79]: ditto

File: crates/sui-json-rpc/src/event_api.rs [hunk 3]
```diff
@@ -58,7 +58,7 @@ impl EventStreamingApiServer for EventStreamingApiImpl {
             event.map(|event| SuiEventEnvelope {
                 timestamp: e.timestamp,
                 tx_digest: e.tx_digest,
-                id: EventID::from((e.seq_num as i64, e.event_num as i64)),
+                id: EventID::from((e.tx_digest.unwrap(), e.event_num as i64)),
```
Comment (patrickkuo) [inline crates/sui-json-rpc/src/event_api.rs:61]: we should make EventEnvelope::tx_digest non-optional, I don't think this is ever None

**Review by patrickkuo — CHANGES_REQUESTED** [review]

File: crates/sui-indexer/src/models/events.rs (hunk 1 above)
Comment (xydas97) [reply crates/sui-indexer/src/models/events.rs:31]: yes, I missed that

File: sdk/typescript/src/providers/json-rpc-provider.ts [hunk 4]
```diff
@@ -621,8 +620,8 @@ export class JsonRpcProvider extends Provider {
   }
 
   async getTransactionDigestsInRange(
-    start: GatewayTxSeqNumber,
-    end: GatewayTxSeqNumber
+    start: TransactionDigest,
```
Comment (patrickkuo) [inline sdk/typescript/src/providers/json-rpc-provider.ts:623]: we shouldn't need to change this...

File: sdk/typescript/src/providers/json-rpc-provider.ts [hunk 5]
```diff
@@ -47,6 +46,7 @@ import {
   CoinMetadataStruct,
   GetObjectDataResponse,
   GetOwnedObjectsResponse,
+  GatewayTxSeqNumber,
```
Comment (patrickkuo) [inline sdk/typescript/src/providers/json-rpc-provider.ts:49]: nit: lets revert all GatewayTxSeqNumber changes in this PR, we don't need to have this in commit history.

**Review by patrickkuo — APPROVED** [review]
just one comment on the GatewayTxSeqNumber changes, otherwise LGTM

**Pull request merged**